src/ansys/grantami/recordlists/_models.py

```diff
@@ -424,7 +434,7 @@ def __init__(
         contains_records_in_databases: Optional[List[str]] = None,
         contains_records_in_integration_schemas: Optional[List[str]] = None,
         contains_records_in_tables: Optional[List[str]] = None,
-        contains_records: Optional[List[str]] = None,
+        contains_records: Optional[List["RecordListItem"]] = None,
```

Comment:
I think this is good. As discussed, we could introduce a `contains_record_guids` where we then build search criteria for each database on the server, which would then preserve the existing functionality, albeit with a different argument name.

I don't have any evidence of anyone wanting to do this search though, so I say we leave it unimplemented unless requested.